File: 02_ProgrammingFundamentals/_06_ObjectsAndClasses/_01_Lecture/_06_Students2.java
package _02_ProgrammingFundamentals._06_ObjectsAndClasses._01_Lecture;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class _06_Students2 {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        String console = scan.nextLine();

        // Add variable
        List<StudentInfo> studentList = new ArrayList<>();

        // Save a list of students and their information
        while (!console.equals("end")) {
            String[] lineFromConsole = console.split("\\s+");

            String firstName = lineFromConsole[0];
            String secondName = lineFromConsole[1];
            int age = Integer.parseInt(lineFromConsole[2]);
            String town = lineFromConsole[3];

            if (!isAlreadyExistingInsideTheList(firstName, secondName, age, town, studentList)) {
                StudentInfo student = new StudentInfo(firstName, secondName, age, town);

                studentList.add(student);
            }

            console = scan.nextLine();
        }

        String town = scan.nextLine();

        // Print result
        for (StudentInfo element : studentList) {
            if (element.getTown().equals(town)) {
                System.out.println(element);
            }
        }

        scan.close();
    }

    // Define method to check if the value already exists in the list
    private static boolean isAlreadyExistingInsideTheList(String firstName, String secondName, int age, String town, List<StudentInfo> studentList) {
        for (StudentInfo element : studentList) {
            boolean firstNameBoolean = element.getFirstName().equals(firstName);
            boolean secondNameBoolean = element.getSecondName().equals(secondName);

            if (firstNameBoolean && secondNameBoolean) {
                element.setAge(age);
                element.setTown(town);

                return true;
            }
        }
        
        return false;
    }
}

// Create practice StudentInfo class
class StudentInfo {
    private String firstName;
    private String secondName;
    private int age;
    private String town;

    public StudentInfo(String firstName, String secondName, int age, String town) {
        this.firstName = firstName;
        this.secondName = secondName;
        this.age = age;
        this.town = town;
    }

    public void setTown(String town) {
        this.town = town;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public String getFirstName() {
        return this.firstName;
    }

    public String getSecondName() {
        return this.secondName;
    }

    public int getAge() {
        return this.age;
    }

    public String getTown() {
        return this.town;
    }

    @Override
    public String toString() {
        return this.firstName + " " + this.secondName + " is " + this.age + " years old";
    }
}

File: 24_C#Fundamentals/_06_ObjectsAndClasses/_01_Lecture/_06_Students2/_06_Students2.cs
using System;

public class _06_Students2
{
    public static void Main(string[] args)
    {
        // Read input value
        string console = Console.ReadLine();

        // Add variable
        List<StudentInfo> studentList = new List<StudentInfo>();

        // Save a list of students and their information
        while (!console.Equals("end"))
        {
            string[] lineFromConsole = console.Split(" ");

            string firstName = lineFromConsole[0];
            string secondName = lineFromConsole[1];
            int age = int.Parse(lineFromConsole[2]);
            string t = lineFromConsole[3]!;

            if (!isAlreadyExistingInsideTheList(firstName, secondName, age, t, studentList))
            {
                StudentInfo student = new StudentInfo(firstName, secondName, age, t);

                studentList.Add(student);
            }

            console = Console.ReadLine();
        }

        string town = Console.ReadLine();

        // Print result
        foreach (StudentInfo element in studentList)
        {
            if (element.Town.Equals(town))
            {
                Console.WriteLine(element);
            }
        }
    }

    // Define method to check if the value already exists in the list
    private static bool isAlreadyExistingInsideTheList(string firstName, string secondName, int age, string town, List<StudentInfo> studentList)
    {
        foreach (StudentInfo element in studentList)
        {
            bool firstNameBoolean = element.FirstName.Equals(firstName);
            bool secondNameBoolean = element.SecondName.Equals(secondName);

            if (firstNameBoolean && secondNameBoolean)
            {
                element.Age = age;
                element.Town = town;

                return true;
            }
        }

        return false;
    }
}

// Create practice StudentInfo class
class StudentInfo
{
    public string FirstName { get; }
    public string SecondName { get; }
    public int Age { get; set; }
    public string Town { get; set; }

    public StudentInfo(string firstName, string secondName, int age, string town)
    {
        this.FirstName = firstName;
        this.SecondName = secondName;
        this.Age = age;
        this.Town = town;
    }

    public override string ToString()
    {
        return $"{this.FirstName} {this.SecondName} is {this.Age} years old.";
    }
}
